code/game/objects/items/twohanded.dm

```diff
@@ -962,3 +963,169 @@
+	if(is_supercharged == 1)
+		var/turf/target_turf = get_turf(target) //Does the nice effects first so whatever happens to what's about to get clapped doesn't affect it
+		var/obj/effect/temp_visual/kinetic_blast/K = new /obj/effect/temp_visual/kinetic_blast(target_turf)
+		K.color = color
+		playsound(loc, 'sound/effects/gravhit.ogg', 80, 1) //Mainly this sound
+		playsound(loc, 'sound/effects/explosion3.ogg', 20, 1) //Bit of a reverb
+		supercharge(user) //At start so it doesn't give an unintentional message if you hit yourself
+		if(istype(target, /obj/machinery))
+			var/obj/machinery/M = target
+			M.take_damage(M.max_integrity*2) //Should destroy machines in one hit
+			if(istype(target, /obj/machinery/door))
+				for(var/obj/structure/door_assembly/D in target_turf) //Will destroy airlock assembly left behind, but drop the parts
+					D.take_damage(D.max_integrity*2)
+			else
+				for(var/obj/structure/frame/F in target_turf) //Will destroy machine or computer frame left behind, but drop the parts
+					F.take_damage(F.max_integrity*2)
+				for(var/obj/structure/light_construct/L in target_turf) //Also light frames because why not
+					L.take_damage(L.max_integrity*2)
+			user.visible_message(span_danger("The hammer thunders against the [target.name], demolishing it!"))
+		if(istype(target, /obj/structure))
+			var/obj/structure/S = target
+			S.take_damage(S.max_integrity*2) //Destroy structures in one hit too
+			if(istype(target, /obj/structure/table))
+				for(var/obj/structure/table_frame/T in target_turf)
+					T.take_damage(T.max_integrity*2) //Destroys table frames left behind
+			user.visible_message(span_danger("The hammer thunders against the [target.name], destroying it!"))
+		if(istype(target, /turf/closed/wall))
+			var/turf/closed/wall/W = target
+			W.dismantle_wall(1) //Deletes the wall but drop the materials, just like destroying a machine above
+			user.visible_message(span_danger("The hammer thunders against the [target.name], shattering it!"))
+			playsound(loc, 'sound/effects/meteorimpact.ogg', 50, 1) //Otherwise there's no sound for hitting the wall, since it's just dismantled
+		if(istype(target, /obj/mecha))
+			var/obj/mecha/M = target
+			M.take_damage(M.max_integrity/3) //A third of its max health is dealt as an untyped damage, in addition to the normal damage of the weapon (which has high AP)
+			user.visible_message(span_danger("The hammer thunders as it massively dents the plating of the [target.name]!"))
+		if(istype(target, /mob/living))
+			var/atom/throw_target = get_edge_target_turf(target, user.dir)
+			var/mob/living/M = target
+			M.throw_at(throw_target, 15, 5) //Same distance as maxed out power fist with three extra force
+			M.Paralyze(20) //Two second stun
+			user.visible_message(span_danger("The hammer thunders as it viscerally strikes [target.name]!"))
+			to_chat(M, span_userdanger("Agony sears through you as [user]'s blow cracks your body off its feet!"))
+			M.emote("scream")
```

Comment:
```suggestion
	if(charging) //Needs a special snowflake check if you hit something that isn't a mob
		if(ismachinery(target) || isstructure(target) || ismecha(target))
			to_chat(user, span_notice("You flip the switch off after your blow."))
			user.visible_message(span_warning("[user] flicks the hammer off after striking [target]!"))
			charging = FALSE
	if(supercharged)
		var/turf/target_turf = get_turf(target) //Does the nice effects first so whatever happens to what's about to get clapped doesn't affect it
		var/obj/effect/temp_visual/kinetic_blast/K = new(target_turf)
		K.color = color
		playsound(loc, 'sound/effects/gravhit.ogg', 80, TRUE) //Mainly this sound
		playsound(loc, 'sound/effects/explosion3.ogg', 20, TRUE) //Bit of a reverb
		supercharge(user) //At start so it doesn't give an unintentional message if you hit yourself
		if(ismachinery(target))
			var/obj/machinery/M = target
			M.take_damage(M.max_integrity * 2) //Should destroy machines in one hit
			if(istype(target, /obj/machinery/door))
				for(var/obj/structure/door_assembly/D in target_turf) //Will destroy airlock assembly left behind, but drop the parts
					D.take_damage(D.max_integrity*2)
			else
				for(var/obj/structure/frame/F in target_turf) //Will destroy machine or computer frame left behind, but drop the parts
					F.take_damage(F.max_integrity*2)
				for(var/obj/structure/light_construct/L in target_turf) //Also light frames because why not
					L.take_damage(L.max_integrity*2)
			user.visible_message(span_danger("The hammer thunders against the [target.name], demolishing it!"))
		if(isstructure(target))
			var/obj/structure/S = target
			S.take_damage(S.max_integrity*2) //Destroy structures in one hit too
			if(istype(target, /obj/structure/table))
				for(var/obj/structure/table_frame/T in target_turf)
					T.take_damage(T.max_integrity*2) //Destroys table frames left behind
			user.visible_message(span_danger("The hammer thunders against the [target.name], destroying it!"))
		if(iswallturf(target))
			var/turf/closed/wall/W = target
			W.dismantle_wall(1) //Deletes the wall but drop the materials, just like destroying a machine above
			user.visible_message(span_danger("The hammer thunders against the [target.name], shattering it!"))
			playsound(loc, 'sound/effects/meteorimpact.ogg', 50, TRUE) //Otherwise there's no sound for hitting the wall, since it's just dismantled
		if(ismecha(target))
			var/obj/mecha/M = target
			M.take_damage(M.max_integrity/3) //A third of its max health is dealt as an untyped damage, in addition to the normal damage of the weapon (which has high AP)
			user.visible_message(span_danger("The hammer thunders as it massively dents the plating of the [target.name]!"))
		if(iisliving(target))
			var/atom/throw_target = get_edge_target_turf(target, user.dir)
			var/mob/living/M = target
			M.throw_at(throw_target, 15, 5) //Same distance as maxed out power fist with three extra force
			M.Paralyze(2 SECONDS)
			user.visible_message(span_danger("The hammer thunders as it viscerally strikes [target.name]!"))
			to_chat(M, span_userdanger("Agony sears through you as [user]'s blow cracks your body off its feet!"))
			M.emote("scream")
```